src/analysis/classic/warlock/demonology/modules/spells/Doomguard.tsx

```diff
@@ -0,0 +1,382 @@
+          );
+        } else {
+          this.snapshotAdvice = (
+            <>
+              You failed to snapshot a buff. Try to summon your doomguard when all buffs are active.
+            </>
+          );
+        }
+        break;
+      default:
+        this.snapshotAdvice = (
+          <>
+            You failed to snapshot several buffs. Try to summon your doomguard when all buffs are
+            active.
+          </>
+        );
+        break;
+    }
+  }
+  private onDoomguardDamage(event: DamageEvent) {
+    if (this.doomguardSummonData.doomguard.totalDamage === undefined) {
+      this.doomguardSummonData.doomguard.totalDamage = 0;
+    }
+    this.doomguardSummonData.doomguard.totalDamage += event.amount;
+  }
+  private onFightEnd(event: FightEndEvent) {
+    if (
+      this.doomguardSummonData.doomguard.summonTimestamp !== 0 &&
+      this.doomguardSummonData.doomguard.deathTimestamp === 0
+    ) {
+      this.doomguardSummonData.doomguard.deathTimestamp = event.timestamp;
+    }
+  }
+  private onDoomguardDeath(event: DeathEvent) {
+    if (event.targetID === this.doomguardId) {
+      this.doomguardSummonData.doomguard.deathTimestamp = event.timestamp;
+      const doomguardDuration =
+        event.timestamp - this.doomguardSummonData.doomguard.summonTimestamp;
+      if (doomguardDuration >= 64000) {
+        this.doomguardSummonData.summonedDemonSummary = (
+          <li>
+            <PerformanceMark perf={QualitativePerformance.Perfect} /> You cast{' '}
+            <SpellLink spell={SPELLS.SUMMON_DOOMGUARD} /> at{' '}
+            {formatDuration(
+              this.doomguardSummonData.doomguard.summonTimestamp - this.owner.fight.start_time,
+            )}
+            , for a full <strong>{formatDuration(doomguardDuration)}</strong> duration and{' '}
+            <strong>{formatNumber(this.doomguardSummonData.doomguard.totalDamage)}</strong> damage{' '}
+          </li>
+        );
+      } else {
+        this.doomguardSummonData.summonedDemonSummary = (
+          <li>
+            <PerformanceMark perf={QualitativePerformance.Good} /> You cast{' '}
+            <SpellLink spell={SPELLS.SUMMON_DOOMGUARD} /> at{' '}
+            {formatDuration(
+              this.doomguardSummonData.doomguard.summonTimestamp - this.owner.fight.start_time,
+            )}
+            , for a <strong>{formatDuration(doomguardDuration)}</strong> duration. It did a total of{' '}
+            {formatNumber(this.doomguardSummonData.doomguard.totalDamage)} damage, but your
+            doomguard did not live for its entire duration. Make sure that cast it soon enough so
+            fight does not end before your doomguard expires.
+          </li>
+        );
+      }
+    }
+  }
+  private onDoomguardCast(event: CastEvent) {
+    this.doomguardSummonData.doomguard.castEvents.push(event);
+    //TODO Check if there's a reliable way to track doomguard not casting for some reasons (being silenced/interrupted/moving/etc.)
```

Comment:
Side note - Doomguard needs Bane of Doom or Bane of Agony on a target to cast

Reply:
I plan to add a cast tracker to be sure if doomguard has downtime, but I still need to work on this to analyze whether downtime is unavoidable or if it is because there is no doom applied /move time / etc. I will handle this later